lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FilePartition.cs

```diff
@@ -0,0 +1,129 @@
+        private readonly ISet<string> _filePaths;
+        private readonly IFileSystem _fileSystem;
+        private readonly IFileSerializer<T> _fileSerializer;
+        private IList<string> localFileNames = new List<string>();
+
+        [Inject]
+        private FilePartition([Parameter(typeof(PartitionId))] string id,
+            [Parameter(typeof(FilePathsInPartition))] ISet<string> filePaths,
+            IFileSystem fileSystem,
+            IFileSerializer<T> fileSerializer)
+        {
+            _id = id;
+            _filePaths = filePaths;
+            _fileSystem = fileSystem;
+            _fileSerializer = fileSerializer;
+
+            if (fileSystem is HadoopFileSystem)
+            {
+                Logger.Log(Level.Info, "+++++HadoopFileSystem");
+            }
+
+            if (fileSystem is LocalFileSystem)
+            {
+                Logger.Log(Level.Info, "!!!!!LocalFileSystem");
+            }
+        }
+
+        public string Id
+        {
+            get { return _id; }
+        }
+
+        /// <summary>
+        /// This method copy remote files to local and then deserialize the files.
+        /// It returns the IEnumerble of T, the details is defined in the Deserialize() method 
+        /// provided by the Serializer
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<T> GetPartitionHandle()
+        {
+            foreach (var f in _filePaths)
+            {
+                Logger.Log(Level.Info, "remoteFileName: " + f);
+                localFileNames.Add(CopyFromRemote(f));
+            }
+
+            return _fileSerializer.Deserialize(localFileNames);
+        }
+
+        private string CopyFromRemote(string sourceFilePath)
+        {
+            Logger.Log(Level.Info, "GetUriPrefix: " + _fileSystem.UriPrefix);
+
+            Uri sourceUri = new Uri(_fileSystem.UriPrefix + sourceFilePath);
+            Logger.Log(Level.Info, "remoteUri: " + sourceUri);
+
+            if (!_fileSystem.Exists(sourceUri))
+            {
+                throw new ApplicationException("Remote File does not exists.");
+            }
+
+            var localFilePath = Path.GetTempPath() + "-partition-" + DateTime.Now.ToString("yyyyMMddHHmmssfff");
+
+            Logger.Log(Level.Info, "localFile: " + localFilePath);
+
+            if (File.Exists(localFilePath))
+            {
+                File.Delete(localFilePath);
+            }
+
+            _fileSystem.CopyToLocal(sourceUri, localFilePath);
+            if (File.Exists(localFilePath))
+            {
+                Logger.Log(Level.Info, "File CopyToLocal!");
+            }
+            else
+            {
+                Logger.Log(Level.Info, "File doesn't CopyToLocal!");               
+            }
+
+            return localFilePath;
+        }
+
+        ~FilePartition()
+        {
+            foreach (var fileName in localFileNames)
+            {
+                File.Delete(fileName);
```

Comment:
Is this safe? Other code might still have a handle on those files? In java, you can register files as temp files which get deleted on process exit. Is there such a thing in C# we can use here?